src/domain/datasets/domain/src/services/dataset_key_value_service.rs

```diff
@@ -16,12 +16,11 @@ use crate::{DatasetEnvVar, DatasetEnvVarNotFoundError, DatasetEnvVarValue};
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-#[async_trait::async_trait]
 pub trait DatasetKeyValueService: Sync + Send {
-    async fn find_dataset_env_var_value_by_key<'a>(
```

Comment:
If API assumes that all vars are extracted and then decrypted on-demand then this method should not be async.

When I see a low-level method that templates a URL string turn `async` - this is a clear alarm bell that we did not separate logic from infrastructure correctly.